File: src/app/interfaces/dialog/OKWindow.java
package app.interfaces.dialog;

import java.net.URL;

import javax.swing.BoxLayout;
import javax.swing.ImageIcon;
import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.WindowConstants;

import java.awt.Component;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import app.telemetry.Logger;
/**
 * @author Jack Meng
 */
public class OKWindow implements ActionListener {
  private final JButton okButton;
  private final JFrame frame;

  public OKWindow(String message) {
    
    JPanel panel = new JPanel();
    JLabel label = new JLabel("<html><center><p>" + message + "</p></center></html>");
    okButton = new JButton("Yay!");
    okButton.addActionListener(this);
    okButton.setAlignmentX(Component.CENTER_ALIGNMENT);
    label.setAlignmentX(Component.CENTER_ALIGNMENT);
    label.setFont(label.getFont().deriveFont(label.getFont().getSize() * 1.5f));

    panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));

    panel.add(label);
    panel.add(okButton);
    URL url = getClass().getResource("/icons/others/ok_icon.png");
    assert url != null;
    ImageIcon icon = new ImageIcon(url);
    frame = new JFrame(message);
    frame.setIconImage(icon.getImage());
    frame.add(panel);
    frame.setSize(300, 100);
    frame.setResizable(false);
    frame.setUndecorated(true);
    frame.addComponentListener(new app.interfaces.event.FrameOrganizer(frame));
    frame.setLocationRelativeTo(null);
    frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
    frame.setVisible(true);
  }

  
  /** 
   * @param e
   */
  @Override
  public void actionPerformed(ActionEvent e) {
    if (e.getSource().equals(okButton)) {
      frame.dispose();
      Logger.log(e.getSource() + "\n" + frame.getClass());
    }
  }

}

File: kt/app/interfaces/dialog/OKWindow.kt
package app.interfaces.dialog

import java.awt.event.ActionListener
import javax.swing.JButton
import javax.swing.JFrame
import java.awt.event.ActionEvent
import javax.swing.JPanel
import javax.swing.JLabel
import javax.swing.BoxLayout
import javax.swing.ImageIcon
import app.interfaces.event.FrameOrganizer
import app.telemetry.Logger
import java.awt.Component
import javax.swing.WindowConstants
import java.io.IOException
import javax.swing.JScrollPane
import java.lang.Runnable

/**
 * @author Jack Meng
 */
class OKWindow(message: String) : ActionListener {
    private val okButton: JButton
    private val frame: JFrame

    /**
     * @param e
     */
    override fun actionPerformed(e: ActionEvent) {
        if (e.source == okButton) {
            frame.dispose()
            Logger.log("""
    ${e.source}
    ${frame.javaClass}
    """.trimIndent())
        }
    }

    init {
        val panel = JPanel()
        val label = JLabel("<html><center><p>$message</p></center></html>")
        okButton = JButton("Yay!")
        okButton.addActionListener(this)
        okButton.alignmentX = Component.CENTER_ALIGNMENT
        label.alignmentX = Component.CENTER_ALIGNMENT
        label.font = label.font.deriveFont(label.font.size * 1.5f)
        panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
        panel.add(label)
        panel.add(okButton)
        val url = javaClass.getResource("/icons/others/ok_icon.png")!!
        val icon = ImageIcon(url)
        frame = JFrame(message)
        frame.iconImage = icon.image
        frame.add(panel)
        frame.setSize(300, 100)
        frame.isResizable = false
        frame.isUndecorated = true
        frame.addComponentListener(FrameOrganizer(frame))
        frame.setLocationRelativeTo(null)
        frame.defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE
        frame.isVisible = true
    }
}